test/test_filesystem.cpp

```diff
@@ -241,6 +244,16 @@ TEST_F(TestFilesystemFixture, is_readable_and_writable) {
     ASSERT_FALSE(nullptr == path);
     EXPECT_TRUE(rcutils_is_readable_and_writable(path));
   }
+  {
+    char * path =
+      rcutils_join_path(this->test_path, "dummy_readable_file.txt", g_allocator);
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      g_allocator.deallocate(path, g_allocator.state);
+    });
+    ASSERT_FALSE(nullptr == path);
```

Comment:
```suggestion
    ASSERT_NE(nullptr, path);
```

Reply:
After private discussion, leaving this as is to match the rest the file.